File: leetcode/120 triangle.js
var minimumTotal = function(triangle) {
    if(triangle.length == 0) return 0;
    
    let [j,sum,lvl] = [0,triangle[0][0],null];
    
    for(let i = 1; i < triangle.length; i++){
        lvl = triangle[i]
        console.log(lvl,j,sum)
        if(lvl[j] >= lvl[j+1]){
            sum += lvl[j+1];
            j++;
        }else{
            sum += lvl[j]; 
        }
    }
    
    return sum;
};

let tri = [
    [2],
    [3, 4],
    [6, 5, 7],
    [4, 1, 8, 3]
];

minimumTotal(tri)

File: leetcode/120 triangle.py
class Solution:
    def minimumTotal(self, triangle: List[List[int]]) -> int:
        f = [0] * (len(triangle) + 1)
        for row in triangle[::-1]: # reversed iteration on the 2d array.
            for i in range(len(row)):
                f[i] = row[i] + min(f[i], f[i + 1])
        return f[0]